src/leiningen/ring/uberwar.clj

```diff
@@ -13,25 +17,54 @@
     (get-cp project)
     (->> (:library-path project) io/file .listFiles (map str))))
 
-(defn contains-entry? [^java.io.File file ^String entry]
-  (with-open [jar-file (JarFile. file)]
-    (some (partial = entry)
-          (map #(.getName ^JarEntry %)
-               (enumeration-seq (.entries jar-file))))))
-
 (defn jar-dependencies [project]
   (for [pathname (get-classpath project)
-        :let [file (io/file pathname)
-              fname (.getName file)]
-        :when (and (.endsWith fname ".jar")
-                   ;; Servlet container will have it's own servlet-api impl
-                   (not (contains-entry? file "javax/servlet/Servlet.class")))]
-    file))
-
-(defn jar-entries [war project]
+        :when (str/ends-with? pathname ".jar")]
+    (io/file pathname)))
+
+(def ^:private javax-servlet? #{"javax/servlet/Servlet.class"})
```

Comment:
Rather than a function, this would be better as a constant. e.g. `servlet-class-resource`.

Reply:
Sorry, I'm not quite following this...do you mean change the name, or the actual value (i.e. the 1-element set)?

Reply:
Something like:

```clojure
(def ^:private `servlet-class-resource` "javax/servlet/Servlet.class")
```

The reason is that `#{servlet-class-resource}` then gives more information to the reader. They know that you're matching against an exact string, whereas `javax-servlet?` is opaque - they'd need to read the definition to understand what it does.

Reply:
ok, I will do this, but bear in mind that I do want to use this as a predicate (twice), so one or another (i.e. here or below), i will have to def that set, so it's not clear to me what the benefit is.  In any case, I'll follow your suggestion...